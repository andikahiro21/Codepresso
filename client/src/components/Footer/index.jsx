import { FormattedMessage } from 'react-intl';

import InstagramIcon from '@mui/icons-material/Instagram';
import FacebookIcon from '@mui/icons-material/Facebook';
import YouTubeIcon from '@mui/icons-material/YouTube';

import classes from './style.module.scss';

const Footer = () => (
  <div className={classes.footer} data-testid="footer">
    <div className={classes.company}>
      <img src="../../static/images/logoNav.png" alt="logo" className={classes.logo} />
      <div className={classes.title}>
        <FormattedMessage id="app_title_header" />
      </div>
    </div>
    <div className={classes.navFooter}>
      <div className={classes.mission}>
        <div className={classes.title}>Our Mission</div>
        <div className={classes.desc}>
          At our company, we strive to make your coffee time more enjoyable. Since 2020, we have been providing
          high-quality coffee and excellent service.
        </div>
        <div className={classes.socCont}>
          <div className={classes.icon}>
            <InstagramIcon />
          </div>
          <div className={classes.icon}>
            <FacebookIcon />
          </div>
          <div className={classes.icon}>
            <YouTubeIcon />
          </div>
        </div>
        <div className={classes.copyright}>2022 Â© Vinicius Okamoto</div>
      </div>
    </div>
  </div>
);

export default Footer;
