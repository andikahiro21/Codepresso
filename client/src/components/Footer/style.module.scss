@import '@styles/base';

.footer {
  @include directionY;
  background-image: url('../../static/images/bg-about.png');
  height: fit-content;
  align-items: center;
  padding-top: 2rem;
  .company {
    @include directionX;
    align-items: center;

    .logo {
      width: 1.5rem;
      height: 2rem;
      margin-right: 0.5rem;
      padding: 0;
    }
    .title {
      font-weight: 800;
      color: $color-white;
      font-size: 2rem;
    }
  }
  .navFooter {
    color: $color-white;
    .mission {
      @include directionY;
      align-items: center;
      text-align: center;
      padding-top: 1.5rem;
      .title {
        display: none;
      }
      .desc {
        width: 70%;
        color: $color-light-grey;
      }
      .socCont {
        @include directionX;
        gap: 0.8rem;
        margin-top: 1.5rem;
      }
      .copyright {
        margin-top: 0.8rem;
        margin-bottom: 2rem;
      }
    }
  }
}
