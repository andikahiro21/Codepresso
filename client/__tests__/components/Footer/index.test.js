import Footer from '@components/Footer';
import { render } from '@utils/testHelper';

let wrapper;

beforeEach(() => {
  wrapper = render(<Footer />);
});

describe('Footer Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('footer')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
