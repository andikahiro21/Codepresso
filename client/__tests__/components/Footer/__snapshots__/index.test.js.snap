// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer"
        data-testid="footer"
      >
        <div
          class="company"
        >
          <img
            alt="logo"
            class="logo"
            src="../../static/images/logoNav.png"
          />
          <div
            class="title"
          >
            Codepresso
          </div>
        </div>
        <div
          class="navFooter"
        >
          <div
            class="mission"
          >
            <div
              class="title"
            >
              Our Mission
            </div>
            <div
              class="desc"
            >
              At our company, we strive to make your coffee time more enjoyable. Since 2020, we have been providing high-quality coffee and excellent service.
            </div>
            <div
              class="socCont"
            >
              <div
                class="icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="InstagramIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                  />
                </svg>
              </div>
              <div
                class="icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FacebookIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                  />
                </svg>
              </div>
              <div
                class="icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="YouTubeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="copyright"
            >
              2022 © Vinicius Okamoto
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer"
      data-testid="footer"
    >
      <div
        class="company"
      >
        <img
          alt="logo"
          class="logo"
          src="../../static/images/logoNav.png"
        />
        <div
          class="title"
        >
          Codepresso
        </div>
      </div>
      <div
        class="navFooter"
      >
        <div
          class="mission"
        >
          <div
            class="title"
          >
            Our Mission
          </div>
          <div
            class="desc"
          >
            At our company, we strive to make your coffee time more enjoyable. Since 2020, we have been providing high-quality coffee and excellent service.
          </div>
          <div
            class="socCont"
          >
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="InstagramIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
            </div>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FacebookIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                />
              </svg>
            </div>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="YouTubeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
                />
              </svg>
            </div>
          </div>
          <div
            class="copyright"
          >
            2022 © Vinicius Okamoto
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
